feature-libs/asm/components/customer-selection/customer-selection.component.ts

```diff
@@ -63,7 +63,7 @@ export class CustomerSelectionComponent implements OnInit, OnDestroy {
   private handleSearchTerm(searchTermValue: string) {
     if (
       Boolean(this.selectedCustomer) &&
-      searchTermValue !== this.selectedCustomer.name
+      searchTermValue !== this.selectedCustomer?.name
```

Comment:
We should not need `?` here, as it's part of the first check. Would also consider dropping `Boolean()` above as it's not solving anything here.

Reply:
Indeed, the first check should be enough.  The compiler does not understand it with `Boolean(this.selectedCustomer)` but does understand it with `!!this.selectedCustomer`.

That's what seemed like the safest change.  It also allows us to remove the extra `?` previously added.